3Delight camera overscan support

gkocov opened the pull request:
Generally describe what this PR will do, and why it is needed

- Camera overscan support (StandardOptions / Overscan ... ) for 3Delight 

This is my first Gaffer pull request, so I hope I have followed the expected conventions and processes correctly. If all is well with this pull request, I have several additional improvements for 3Delight's support in Gaffer that I have already implemented and I'll create individual pull requests for each of these additional features. 

### Checklist ###

- [x] I have read the [contribution guidelines](https://github.com/GafferHQ/gaffer/blob/main/CONTRIBUTING.md).
- [x] I have updated the documentation, if applicable.
- [x] I have tested my change(s) in the test suite, and added new test cases where necessary.
- [x] My code follows the Gaffer project's prevailing coding style and conventions.

johnhaddon commented on src/IECoreDelight/Renderer.cpp:
```diff
@@ -1505,6 +1505,24 @@ class DelightRenderer final : public IECoreScenePreview::Renderer
 			const V2i &resolution = camera->getResolution();
 			screeenParameters.add( { "resolution", resolution.getValue(), NSITypeInteger, 2, 1, NSIParamIsArray } );
 
+			const int overscanLeft = static_cast<int>(camera->getOverscanLeft() * resolution.x);
```
The Camera class also has a [`getOverscan()`](https://github.com/ImageEngine/cortex/blob/main/include/IECoreScene/Camera.h#L170) method that returns a boolean to say whether or not overscan should be active. We need to check that it is on before applying it.

gkocov replied:
This should be fixed now. I added a check to see if getOverscan() is true or false and depending on that add the overscan parameters to the NSI screen.

johnhaddon commented on src/IECoreDelight/Renderer.cpp:
```diff
@@ -1505,6 +1505,24 @@ class DelightRenderer final : public IECoreScenePreview::Renderer
+
 			Box2i renderRegion = camera->renderRegion();
 
 			// I can't find any support in 3delight for overscan - and if crop goes outside 0 - 1,
```
This old comment seems to be inaccurate now. It also seems that the old handling of crop windows isn't consistent with the new overscan code (crop is intended to be measured with respect to the non-overscanned image), which you can see if you compare to an Arnold render. I think fixing that could be a bit involved so let's not tackle it in this PR - what you've done already represents good progress.

Could you just update the old comment to say this?

```
// If crop goes outside 0 - 1, 3Delight ignores it, so we clamp.
/// \todo 3Delight interprets crop as 0-1 across the overscanned data window, but
/// Gaffer defines it as 0-1 across the original (non-overscanned) display window.
/// Adjust for that. This will only work nicely with the CropWindowTool if we also
/// update the display driver to output an accurate display window for overscanned renders.
```

gkocov replied:
Absolutely no problem. Comment updated.

johnhaddon reviewed (commented):
Thanks Goran, and congrats on your first full request! Gaffer's 3Delight support has always lagged well behind our Arnold and Cycles support, so its brilliant to see some progress on this front - I'm looking forward to your other PRs too.

I've made a couple of comments inline that should hopefully be easy to address. It would also be great if you could add a unit test for your new functionality - you can see the current camera test here :
https://github.com/GafferHQ/gaffer/blob/main/python/IECoreDelightTest/RendererTest.py#L445.

Thanks! 
Cheers...
John

gkocov commented:
Thanks for the kind words John! I updated the overscan code and comment per your suggestions and I also added the overscan unit test. If there are any additional changes needed please let me know. I'll start on the other PRs in the mean time.
